pkg/api/const.go

```diff
@@ -98,6 +98,8 @@ const (
 	ScaleSetPriorityRegular = "Regular"
 	// ScaleSetPriorityLow means the ScaleSet will use Low-priority VMs
 	ScaleSetPriorityLow = "Low"
+	// ScaleSetPrioritySpot means the ScaleSet will use Spot VMs
+	ScaleSetPrioritySpot = "Spot"
```

Comment:
Let's document the new value. https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#L567 is a good place to start (there's already an entry for ScaleSetPriority).

Reply:
Thanks. Edited the doc and added explanation to the new field.

Reply:
Everyone probably already knows this - but "Low" will be deprecated at one year from now and no longer work

Reply:
Thanks @jluk that's a good point, maybe we should add a note to the docs with the link to the deprecation notice, right now it looks like we're equally recommending both options.

Reply:
perhaps, even a warning for api models which include low.

Reply:
Slight preference for removing low as an option altogether (w/ back-compat support for pre-existing cluster). (As a distinct, follow-up PR)

If we can prevent any aks-engine created clusters from running low pri vmss a year from now when it is officially deprecated, that would be 💯

Reply:
+1 on removing the option in a follow up PR. We might not have time to get to the follow up right away though so in the meantime adding a small note to the docs about deprecation is a low hanging fruit.

Reply:
Added "(DEPRECATED)" notation in the doc in this PR after the "Low" option. @jluk VMSS team mentioned yesterday that they will treat "Low" as "Spot" VMSS internally. So it's better to remove it early as "Low" does not give the option to set maxPrice in AKS-E.

Reply:
Stopping more clusters from setting low sounds fine and I'll defer how to language that into engine.

For a year low is still accepted and just behaves as spot with maxPrice = -1 so not truly deprecated yet just on the path to it. Also Jag is owns Spot, not VMSS so these rules apply for both VM and VMSS.